### PR TITLE
Fix KeywordManager initialization

### PR DESCRIPTION
**Pull request by aalmada**
A fix to issue #396 and removed Linq.

### CODE REVIEW

File: Assets/HoloToolkit/Input/Scripts/Voice/KeywordManager.cs [hunk 1]
```diff
@@ -44,23 +44,36 @@ public enum RecognizerStartBehavior { AutoStart, ManualStart };
...
-                keywordRecognizer.OnPhraseRecognized += KeywordRecognizer_OnPhraseRecognized;
-
-                if (RecognizerStart == RecognizerStartBehavior.AutoStart)
+                try
                 {
-                    keywordRecognizer.Start();
+                    string[] keywords = new string[keywordCount];
+                    // Convert the struct array into a dictionary, with the keywords and the keys and the methods as the values.
+                    // This helps easily link the keyword recognized to the UnityEvent to be invoked.
+                    for (int index = keywordCount; --index >= 0;)
```
Comment (keveleigh) [inline Assets/HoloToolkit/Input/Scripts/Voice/KeywordManager.cs]: Is LINQ "bad" enough to remove it from a method that only gets called once (or maybe a handful of times, if you have more than one KeywordManager) in an app's lifetime? This code seems slightly less readable to me, though I'm all good with replacing this if I have a minority opinion!

Comment (aalmada) [inline Assets/HoloToolkit/Input/Scripts/Voice/KeywordManager.cs]: I believe Linq can be "naughty" in Update() but not in Start(). I also believe that var, Linq and Reactive Extensions are more readable, once you get used to. I'm just following the project's rules here. ;)
In this case, there's a small performance gain as the array is iterated only once to fill both the dictionary and the keywords array.
Avoidance of 'foreach' has also been one of the common suggestions so I decided to try the reverse order for loop with prefixed decrement, that I usually use in C++ but never in C#. Foreach loop is [very performant on arrays](http://stackoverflow.com/questions/365615/in-net-which-loop-runs-faster-for-or-foreach/472258#472258).

Comment (aalmada) [inline Assets/HoloToolkit/Input/Scripts/Voice/KeywordManager.cs]: I run a [benchmark](https://gist.github.com/aalmada/b535b09ed83092b9288ff55d59b3b656) and to iterate an array of 100000000 ints, it takes:
foreach: 972ms
for: 742ms
reversed for: 566ms
It confirms my experience but, in this case, the number of keywords is MUCH smaller so readability may prevail. Should I change it? To a foreach or a for?

Comment (aalmada) [inline Assets/HoloToolkit/Input/Scripts/Voice/KeywordManager.cs]: I think the foreach is reasonable enough for this case.

Comment (aalmada) [inline Assets/HoloToolkit/Input/Scripts/Voice/KeywordManager.cs]: I realized just now that I need the index to assign to the keywords array. I'll have to leave it as a for loop but I'll change it to incremental order.
Highly performing code is usually not elegant...

**Comment by keveleigh** [comment]
Looks good! Thanks!

**Pull request merged**